### ISSUE TITLE
Mute button should not be bound by default

### ISSUE THREAD

**Issue by Desour** [minetest/minetest#13318]
### Problem

The mute button is by default bound to m. It's easy to press this button by accident, resulting in the player wondering why the sound doesn't work.

### Solution

Don't bind the mute button by default.

Opinions?

### Additional context

See:
https://github.com/minetest/minetest/blob/c91f3f99fdd81e37527b7208fbf0b6287421d779/src/defaultsettings.cpp#L103

**Comment by Montandalar** [comment]
This wouldn't be such a big issue if Minetest had a menu structured like a lot of games with an audio section to let you know that the game had been muted, but there's basically no way to work out that if Minetest is muted by using the Esc menu. So it's probably for the best.

**Comment by Montandalar** [comment]
> Minetest has that: Esc -> Sound volume -> checkbox for muted

Oh you're right. Blindness sets in from too much familiarity I think. I'm not really in favour of unbinding the key in that case. It's not as unsemantic, easy to accidentally push as pitch move or infinite render distance, or unexplainable from a noob perspective. "No sound?" in chat -> tell them to check if it's muted in the menu.

**Comment by Wuzzy2** [comment]
I find a 1-keypress for muting and unmuting the sound to be actually useful.

**Comment by farooqkz** [comment]
> I find a 1-keypress for muting and unmuting the sound to be actually useful.

And I've found the opposite. I do PvP and there is no reason I must mute the sound in combat accidentally. There is a third solution however: Keep the current binding but show some notification in the game and tell the user how to unmute. Maybe even a constant text telling the player that it is muted and how to unmute?

**Comment by Montandalar** [comment]
>> And I've found the opposite. I do PvP and there is no reason I must mute the sound in combat accidentally. There is a third solution however: Keep the current binding but show some notification in the game and tell the user how to unmute. Maybe even a constant text telling the player that it is muted and how to unmute?

I strongly disagree with an icon, text or similiar - not unless we also had status icons for things like fly, fast, noclip pitch move, autoforward and so on. As a PvPer, you should be prepared to rebind or clear keybinds. That's the nature of that kind of gameplay. And I'm not even sure how you would be accidentally pushing m, given it's nowhere near WASD, nor is it on the same row as the HJK keys for noclip, fast & fly.

**Comment by farooqkz** [comment]
> > And I've found the opposite. I do PvP and there is no reason I must mute the sound in combat accidentally. There is a third solution however: Keep the current binding but show some notification in the game and tell the user how to unmute. Maybe even a constant text telling the player that it is muted and how to unmute?
> 
> I strongly disagree with an icon, text or similiar - not unless we also had status icons for things like fly, fast, noclip pitch move, autoforward and so on. As a PvPer, you should be prepared to rebind or clear keybinds. That's the nature of that kind of gameplay. And I'm not even sure how you would be accidentally pushing m, given it's nowhere near WASD, nor is it on the same row as the HJK keys for noclip, fast & fly.

I agree there is low probability for me to accidentally press the m button. But if I remember correctly, it wasn't the case in the first days of minetesting which was years ago. I, however, disagree that (un)mute key binding must be 1-keystroke out there and by default. I personally don't use it and I think this is the case for most other players. But if there are users who use it frequently and need to be just 1 keystroke away from (un)mute, they could enable it in Settings.

As an additional note, adding icons or something for other keys(HJK) will definitely be beginner friendly. In my early days of Minetest I used to press them accidentally, too. However they usually require their own permissions so no instant effect most of the time.

**Comment by rubenwardy** [comment]
I personally just use the mute key on my keyboard when I need to mute it. But not every player has one. Do we get many complaints about the sound not working?

**Comment by farooqkz** [comment]
In my experience, most complain won't complain when it has arrived into their hands for free of charge. They will simply uninstall it. So we can't be sure about this just by counting number of people who complain.

**Comment by Desour** [comment]
>Do we get many complaints about the sound not working?

I haven't seen any. This issue was created because of personal experiences.

----

Opinions so far (with username starting letters):
* unbind default: 3 (D, f, r)
* keep default: 4 (M, W, Z, h)

**Comment by Desour** [comment]
If we add a hud element if the game is muted, mods must be able to move it or disable it completely. (Or they'd even have to enable it explicitly.)
A question is where we would put it by default.

We could also put a mute symbol, or "[muted]" into the window title, or replace the minetest logo with one where a mute image is overlayed.

I don't think we need status icons for fly, noclip or fast. Players can easily notice if it's enabled.

**Comment by Montandalar** [comment]
![screenshot_20230315_141143](https://user-images.githubusercontent.com/13659694/225195950-485ce5ea-cb85-4d7c-af4c-7c19999947c8.png)

Here's an antagonistic scenario screenshot, with chat, the minimap, areas mod, dreambuilder hotbar extender, and postool all showing. Those are all fairly common mods, and maybe those are the ones that can move. But still, where would  you place your hypothetical mute icon?

I am inclined towards Desour's idea of [muted] in the title bar. It may not be visible for fullscreen players, though I don't know what fraction of players that it. I basically never play Minetest full screen unless taking a good screenshot or  screen recording. And I feel like I've seen it in another game before.

**Comment by hlqkj** [comment]
The tile bar text and application icon are usually OS business, playing with them could mean potentially "weird" results in some systems, now and/or in future. What if the OS adds an own overlay, for example?

Granted that the discussion deviated a bit from the original title, I think that using the "if you add ... where would you put it?" as a reason to support/discourage a certain feature is pointless.

We already have temporary overlays, or notifications, for some status changes (again: fly, fast, noclip, mute) so if something has to be introduced about the muted one, why not just start from what we already have? That is, make the mute notification a separate, maybe slightly higher overlay text that stays on as long as the audio is muted?

The HUD clogging problem is there, that is true. But... First of all, a static overlay feature for the muted state would be redundant and hence pointless in android. Unless my memory fails me, there are status icons in a menu there. So, the change would only affect PC users, which reasonably have more space than a mobile device.

Then on PC, it is _not_ engine's fault if, over time, mod devs abused the HUD without any planning or agreement on reccommended practices to follow. I've seen servers where areas+protector would fill half of the screen, or games offering an insanely huge hotbar (whereas half of it can't even be used in mobile devices). So, can this be a reason to restrict the engine development?...

Finally, about giving mods the ability to "disable it completely". This is pointless, or an example of over engineering at best. Unlike for example the debug info. or minimap, a "Muted" overlay could in no way give an unfair advantage to players, or be altering the gameplay so heavily for it to be forced off. Maybe it could be useful if it can be moved, but then what about other notifications? (Fly/fast/etc.)

**Comment by Montandalar** [comment]
> The tile bar text and application icon are usually OS business, playing with them could mean potentially "weird" results in some systems, now and/or in future. What if the OS adds an own overlay, for example?
A lot of applications do it though, like adding notification icons to an instant messaging application. It would be a headache to manage though from a technical standpoint I think.

> Then on PC, it is _not_ engine's fault if, over time, mod devs abused the HUD without any planning or agreement on reccommended practices to follow. I've seen servers where areas+protector would fill half of the screen, or games offering an insanely huge hotbar (whereas half of it can't even be used in mobile devices). So, can this be a reason to restrict the engine development?...
I think you make a great point here, but I would still like to see someone propose where exactly they think it should go. I'm thinking right-aligned under the minimap personally - yes, dynamically positioned.

>hence pointless in android. Unless my memory fails me, there are status icons in a menu there.
Good point. So the HUD flag would be turned off by default or configured differently for mobile builds just like how the crosshair is hidden for touch contorls.

> Finally, about giving mods the ability to "disable it completely".
This is just being consistent with every other HUD in the engine: the hotbar, minimap, radar and so on. Minetest is an engine, and game authors deserve complete control over their games' UIs. They can leave the engine's mute icon as-is, reskin it, re-engineer it, throw it out, whatever they please. We already have the means, just make it another HUD flag.

To sum up my current view: I think Minetest should add a mute icon below the minimap that shows when the game is muted. It should be configurable in the menu and through game HUD flags like any other HUD. It should not be present on Android. And the keybinding can stay.

**Comment by MisterE123** [comment]
the suggestion is to unbind [m] by default. @Wuzzy2 why can't you just rebind it in your copy?

**Comment by Desour** [comment]
> the suggestion is to unbind [m] by default. @Wuzzy2 why can't you just rebind it in your copy?

There are currently more people for keeping the default bind. So, the ones who don't want the bind should unbind it.
(Though I currently don't know how unbinding can be done with the gui...)
